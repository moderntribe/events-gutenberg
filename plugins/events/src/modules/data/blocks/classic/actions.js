/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducers';

export const setDetailsTitle = ( title ) => ( {
	type: types.SET_CLASSIC_DETAILS_TITLE,
	payload: {
		title,
	},
} );

export const setOrganizerTitle = ( title ) => ( {
	type: types.SET_CLASSIC_ORGANIZERS_TITLE,
	payload: {
		title,
	},
} );

export const setInitialState = ( { get } ) => ( dispatch ) => {
	dispatch( setDetailsTitle( get( 'detailsTitle', DEFAULT_STATE.detailsTitle ) ) );
	dispatch( setOrganizerTitle( get( 'organizerTitle', DEFAULT_STATE.organizerTitle ) ) );
};
