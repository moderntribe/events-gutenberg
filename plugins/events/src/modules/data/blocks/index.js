/**
 * Internal dependencies
 */
import reducer from './reducers';

export default reducer;
