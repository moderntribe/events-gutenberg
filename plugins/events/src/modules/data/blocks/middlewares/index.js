export * from './dates';
export { default as multiDay } from './multiday';
export { default as allDay } from './allday';
export * from './time';
