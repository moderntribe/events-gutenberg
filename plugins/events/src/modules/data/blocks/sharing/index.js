/**
 * Internal dependencies
 */
import reducer from './reducers';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types };
