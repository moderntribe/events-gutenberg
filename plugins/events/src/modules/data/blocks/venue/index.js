/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils };
