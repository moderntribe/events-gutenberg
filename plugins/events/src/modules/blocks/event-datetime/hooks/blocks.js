/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Internal Dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';
import { PluginBlockHooks } from '@moderntribe/common/components';

const PLUGIN_TEMPLATES = {
	[ constants.EVENTS_PRO_PLUGIN ]: [
		[ 'tribe/event-pro-recurring', {}],
	],
};

const PluginDateTimeBlockHooks = () => {
	return (
		<PluginBlockHooks
			pluginTemplates={ PLUGIN_TEMPLATES }
			templateLock="all"
		/>
	);
};

PluginDateTimeBlockHooks.propTypes = {};

export default PluginDateTimeBlockHooks;
