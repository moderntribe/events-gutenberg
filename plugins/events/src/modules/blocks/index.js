import { registerBlockType } from '@wordpress/blocks';

import classicEventDetails from 'blocks/classic-event-details';

import eventDateTime from 'blocks/event-datetime';
import eventVenue from 'blocks/event-venue';
import eventOrganizer from 'blocks/event-organizer';
import eventLinks from 'blocks/event-links';
import eventPrice from 'blocks/event-price';
import eventCategory from 'blocks/event-category';
import eventTags from 'blocks/event-tags';
import eventWebsite from 'blocks/event-website';
import FeaturedImage from 'blocks/featured-image';
import { initStore } from 'data';

import './style.pcss';

initStore();

const blocks = [
	classicEventDetails,

	eventDateTime,
	eventVenue,
	eventOrganizer,
	eventLinks,
	eventPrice,
	eventCategory,
	eventTags,
	eventWebsite,
	FeaturedImage,
];

blocks.forEach( block => {
	const blockName = `tribe/${ block.id }`;
	registerBlockType( blockName, block );
} );

export default blocks;

