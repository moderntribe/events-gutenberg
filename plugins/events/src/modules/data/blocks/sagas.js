
/**
 * External dependencies
 */
import { store } from '@moderntribe/common/store';

/**
 * Internal dependencies
 */
import * as price from '@moderntribe/events/data/blocks/price';
import * as venue from '@moderntribe/events/data/blocks/venue';

[
	price.sagas,
	venue.sagas,
].forEach( sagas => store.run( sagas ) );
