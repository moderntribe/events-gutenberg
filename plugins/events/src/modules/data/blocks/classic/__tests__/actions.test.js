/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@@plugins/events/data/blocks/classic';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Classic actions', () => {
	test( 'Action to set the organizer title', () => {
		expect( actions.setOrganizerTitle( 'Modern Tribe' ) ).toMatchSnapshot();
	} );

	test( 'Action to set the details title', () => {
		expect( actions.setDetailsTitle( 'Events' ) ).toMatchSnapshot();
	} );
} );

describe( '[STORE] - Classic thunk actions', () => {
	test( 'Action to set the initial state', () => {
		const store = mockStore();
		const get = jest.fn( ( key ) => {
			const values = {
				detailsTitle: 'Events',
				organizerTitle: 'Modern Tribe',
			};
			return values[ key ];
		} );

		store.dispatch( actions.setInitialState( { get } ) );

		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 2 );
		expect( store.getActions() ).toMatchSnapshot();
	} );
} );
