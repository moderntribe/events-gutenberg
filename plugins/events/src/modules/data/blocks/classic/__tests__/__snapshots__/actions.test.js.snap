// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Classic actions Action to set the details title 1`] = `
Object {
  "payload": Object {
    "title": "Events",
  },
  "type": "@@MT/EVENTS/SET_CLASSIC_DETAILS_TITLE",
}
`;

exports[`[STORE] - Classic actions Action to set the organizer title 1`] = `
Object {
  "payload": Object {
    "title": "Modern Tribe",
  },
  "type": "@@MT/EVENTS/SET_CLASSIC_ORGANIZERS_TITLE",
}
`;

exports[`[STORE] - Classic thunk actions Action to set the initial state 1`] = `
Array [
  Object {
    "payload": Object {
      "title": "Events",
    },
    "type": "@@MT/EVENTS/SET_CLASSIC_DETAILS_TITLE",
  },
  Object {
    "payload": Object {
      "title": "Modern Tribe",
    },
    "type": "@@MT/EVENTS/SET_CLASSIC_ORGANIZERS_TITLE",
  },
]
`;
